From: Roger Shimizu <rogershimizu@gmail.com>
Date: Tue, 21 Jun 2016 22:20:06 +0900
Subject: fix a bit on library pkg-config file

---
 src/libcork.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libcork.pc.in b/src/libcork.pc.in
index 1f12376..9b61bb0 100644
--- a/src/libcork.pc.in
+++ b/src/libcork.pc.in
@@ -3,7 +3,7 @@ exec_prefix=${prefix}
 libdir=${exec_prefix}/@CMAKE_INSTALL_LIBDIR@
 includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 sharedir=${prefix}/share
-sphinxdir=${sharedir}/doc/libcork/html
+sphinxdir=${sharedir}/doc/libcork-doc/html
 
 Name: libcork
 Description: A simple, easily embeddable cross-platform C library
