From: Roger Shimizu <rogershimizu@gmail.com>
Date: Thu, 6 Oct 2016 01:31:16 +0900
Subject: make version in soname the same as library version

Bug: https://lists.debian.org/debian-sparc/2016/08/msg00035.html
---
 cmake/FindCTargets.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/FindCTargets.cmake b/cmake/FindCTargets.cmake
index 455d428..b5d62fe 100644
--- a/cmake/FindCTargets.cmake
+++ b/cmake/FindCTargets.cmake
@@ -63,7 +63,7 @@ function(add_c_library __TARGET_NAME)
         message(FATAL_ERROR "Invalid library version number: ${__VERSION}")
     endif (__VERSION MATCHES "^([0-9]+)\\.([0-9]+)\\.([0-9]+)(-dev)?$")
 
-    math(EXPR __SOVERSION "${__VERSION_CURRENT} - ${__VERSION_AGE}")
+    set(__SOVERSION "${__VERSION_CURRENT}")
 
     get_property(ALL_LOCAL_LIBRARIES GLOBAL PROPERTY ALL_LOCAL_LIBRARIES)
     list(APPEND ALL_LOCAL_LIBRARIES ${__TARGET_NAME})
